platforms/jvm/testing-jvm/src/integTest/groovy/org/gradle/testing/junit/junit4/JUnit4AssumptionFailureIntegrationTest.groovy

```diff
@@ -0,0 +1,134 @@
+        """
+        file("src/test/java/com/example/MyTest.java") << """
+            package com.example;
+
+            import org.junit.Assume;
+            import org.junit.Test;
+
+            public class MyTest {
+                @Test
+                public void theTest() {
+                    Assume.assumeTrue("skipped reason", false);
+                }
+            }
+        """
+        when:
+        succeeds("test")
+        then:
+        outputContains("Assumption failure: skipped reason")
+        def testResult = new DefaultTestExecutionResult(testDirectory)
+        testResult.testClass("com.example.MyTest").assertTestSkipped("theTest") {
+            assert it.message == "skipped reason"
+            assert it.type == "org.junit.AssumptionViolatedException"
+            assert it.text.contains("skipped reason")
+        }
+    }
+
+    def "does not capture ignored tests as assumption failures"() {
+        buildFile << """
+            plugins {
+                id 'java-library'
+            }
+            ${mavenCentralRepository()}
+
+            testing {
+                suites {
+                    test {
+                        useJUnit()
+                        targets {
+                            all {
+                                testTask.configure {
+                                    addTestListener(new TestListener() {
+                                        void beforeSuite(TestDescriptor suite) {}
+                                        void afterSuite(TestDescriptor suite, TestResult result) {}
+                                        void beforeTest(TestDescriptor testDescriptor) {}
+                                        void afterTest(TestDescriptor testDescriptor, TestResult result) {
+                                            println("No assumption failure")
+                                            assert result.assumptionFailure == null
+                                        }
+                                    })
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        """
+        file("src/test/java/com/example/MyTest.java") << """
+            package com.example;
+
+            import org.junit.Ignore;
+            import org.junit.Test;
+
+            public class MyTest {
+                @Test
+                @Ignore
+                public void theTest() {
+                    // fail the test if it actually runs
+                    assert false;
+                }
+            }
+        """
+        when:
+        succeeds("test")
+        then:
+        outputContains("No assumption failure")
+        def testResult = new DefaultTestExecutionResult(testDirectory)
+        testResult.testClass("com.example.MyTest").assertTestSkipped("theTest") {
+            assert it.message.isEmpty()
+            assert it.type.isEmpty()
+            assert it.text.isEmpty()
```

Comment:
🤔 Could we get "Test marked with @Ignore" into the message here somehow?

Reply:
Not really because we don't have that information from the testing framework.

Reply:
Odd that JUnit doesn't support that, considering the let/encourage you to put a value in the annotation to note why something was ignored.